catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -25,23 +26,30 @@
 import org.apache.commons.lang.StringUtils;
 import org.codice.ddf.catalog.async.data.api.internal.InaccessibleResourceException;
 import org.codice.ddf.catalog.async.data.api.internal.ProcessResource;
+import org.codice.ddf.platform.util.TemporaryFileBackedOutputStream;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class LazyProcessResourceImpl implements ProcessResource {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(LazyProcessResourceImpl.class);
 
+  private static final int FILE_BACKED_OUTPUT_STREAM_THRESHOLD = 32 * 1024; // 32 kilobytes
+
   private URI uri;
 
-  private String name = DEFAULT_NAME;
+  private volatile String name = DEFAULT_NAME;
```

Comment:
It prevents having to synchronize the getters for no other reasons

Reply:
Understood, but two of the three getters first invoke a synchronized method that sets the values before returning anyway. So for at least `name` and `mimeType`, it's extra noise (if not any less efficient).

Reply:
I missed that. Than I agree with you. Let's remove volatile and move all of the getters that access these variables to be synchronized

Reply:
That makes sense to me, but I'm not too keen on concurrency issues so when @paouelle suggested I make those volatile I trusted his wisdom. 

As for the asymmetry in the getters, I think one goal of this class was to avoid loading any other variables if only the size and/or uri are needed so those variables can be set and then retrieved without loading anything else. I'm not sure, however, what makes loading those other variables so resource intensive that we need to avoid it.

Reply:
Gotcha. I would then leave `size` as volatile and make the `getName()` and `getMimeType()` methods synchronized (since they're sort of implicitly sync by calling `loadResource()` anyway). This way, it'll be, I think, a little clearer to those maintaining down the road why things are the way they are.